Fix an error where relationships would not be properly deserialized

Battlesheepu opened the pull request:
Fix an error from issue #127  where multi-level nested relationships do not serialize as full objects, even if 'schema' parameter is passed when creating a Relationship field

Battlesheepu committed:
Fix an error where relationships would not be properly deserialized from included data, even if 'schema' parameter is passed when creating a Relationship field.

Battlesheepu committed:
Fix Python 2.7 compatibility in the tests, where next() was used on a list (since filter returns a list in Python 2.7).

Battlesheepu committed:
Reduce line length in accordance with PEP8.

sloria commented on tests/test_schema.py:
```diff
@@ -683,6 +683,65 @@ def test_deserializing_relationship_fields(self):
+        ]
+
+        included_author = filter(lambda item: item['type'] == 'people', article['included'])
+        included_author = list(included_author)[0]
```
Minor: I believe this can be simplified to 

```python
included_author = next(filter(lambda item: item['type'] == 'people', article['included']))
```

sloria reviewed (changes requested):
Sorry for the delay in reviewing this.

Can you please get this up to date with the latest `dev` branch?

It appears that the test is failing when using marshmallow 3.

Battlesheepu committed:
Merge branch 'dev' into fix-nested-relationships-deserialization

# Conflicts:
#	AUTHORS.rst

Battlesheepu committed:
Simplify test code by merging two lines into one

Battlesheepu committed:
Fix incorrect styling

Battlesheepu committed:
Revert: Simplify test code by merging two lines into one

Battlesheepu committed:
Fix tests by adding missing author field, required for provided data due to the strict=True flag

Battlesheepu commented:
Thanks for the review, @sloria , I've updated the code to the current state of `dev` branch. 

The problem was the `author` field - present in the article's test data, but missing from my test schema. The tests failed with or without my fix when using a test schema with `strict=True`. I've added the field and the tests are all green.

As for the minor change you've requested - it's incompatible with Python 2.7. Sorry for the mess, since first I've commited the change you've proposed, but later had to revert it anyway.

sloria commented:
Thanks @ww3pl for getting this up to date. This is released in 0.20.4.

By the way, I'm starting to use something along the lines of the [Pull Request Hack](http://felixge.de/2013/03/11/the-pull-request-hack.html). Basically, if you get a PR merged in, you get an invite to have commit access.

No pressure to do any more work than you already have. Just want to open the door to further collaboration. =)
